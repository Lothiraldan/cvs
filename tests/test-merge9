#!/bin/sh

# test that we don't interrupt the merge session if
# a file-level merge failed

hg init repo
cd repo

echo foo > foo
echo a > bar
hg ci -Am 'add foo' -d '0 0'

hg mv foo baz
echo b >> bar
echo quux > quux1
hg ci -Am 'mv foo baz' -d '0 0'

hg up -qC 0
echo >> foo
echo c >> bar
echo quux > quux2
hg ci -Am 'change foo' -d '0 0'

# test with the rename on the remote side
HGMERGE=false hg merge
hg resolve -l

# test with the rename on the local side
hg up -C 1
HGMERGE=false hg merge

echo % show unresolved
hg resolve -l

echo % unmark baz
hg resolve -u baz

echo % show
hg resolve -l

echo % re-resolve baz
hg resolve baz

echo % after
hg resolve -l

echo % resolve all
hg resolve

echo % after
hg resolve -l

true
