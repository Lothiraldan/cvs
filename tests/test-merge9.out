adding bar
adding foo
adding quux1
adding quux2
created new head
merging bar
merging bar failed!
merging foo and baz to baz
1 files updated, 1 files merged, 0 files removed, 1 files unresolved
use 'hg resolve' to retry unresolved file merges or 'hg update -C' to abandon
U bar
R baz
3 files updated, 0 files merged, 1 files removed, 0 files unresolved
merging bar
merging bar failed!
merging baz and foo to baz
1 files updated, 1 files merged, 0 files removed, 1 files unresolved
use 'hg resolve' to retry unresolved file merges or 'hg update -C' to abandon
% show unresolved
U bar
R baz
% unmark baz
% show
U bar
U baz
M bar
M baz
M quux2
? bar.orig
% re-resolve baz
merging baz and foo to baz
% after
U bar
R baz
% resolve all warning
abort: no files or directories specified; use --all to remerge all files
% resolve all
merging bar
warning: conflicts during merge.
merging bar failed!
% after
U bar
R baz
